templates/analysis.yml

```diff
@@ -40,84 +26,70 @@ parameters:
-    displayName: Build ${{ parameters.microservice }} --> Task
-
-  - task: SonarCloudAnalyze@1
-    displayName: Analyze ${{ parameters.microservice }} --> Task
-# End of Angular-specific Analysis Steps --> if
-
-- ${{ if eq(parameters.angular, false) }}:  # <-- Maven build is the default
-  - task: SonarCloudPrepare@1 # <-- See external requirements note above
-    inputs:
-      SonarCloud: 'SonarCloud'
-      organization: ${{ parameters.sonarOrg }}
-      scannerMode: 'Other'
-      extraProperties: |
-        sonar.projectKey=${{ parameters.sonarKeyPrefix }}${{ parameters.microservice }}
-        sonar.projectName=${{ parameters.sonarNamePrefix }}${{ parameters.microservice }}
-    displayName: Prepare Analysis for ${{ parameters.microservice }} --> Task
+- task: SonarCloudPrepare@1 # <-- See external requirements note above
+  inputs:
+    SonarCloud: 'SonarCloud'
+    organization: ${{ parameters.sonarOrg }}
+    scannerMode: 'Other'
+    extraProperties: |
+      sonar.projectKey=${{ parameters.sonarKeyPrefix }}${{ parameters.microservice }}
+      sonar.projectName=${{ parameters.sonarNamePrefix }}${{ parameters.microservice }}
+  displayName: Prepare Analysis for ${{ parameters.microservice }} --> Task
 
-  - task: 'Maven@3'
-    inputs:
-      mavenPomFile: ${{ parameters.workDir }}pom.xml
-      publishJUnitresults: true
-      testResultsFiles: '**/surefire-reports/TEST-*.xml'
-      codeCoverageToolOption: 'JaCoCo'
-      codeCoverageClassFilesDirectories: ${{ parameters.coveragePath }}
-      javaHomeOption: 'JDKVersion'
-      jdkVersionOption: '1.11'
-      mavenVersionOption: 'Default'
-      mavenOptions: '-Xmx3072m'
-      mavenAuthenticateFeed: false
-      effectivePomSkip: false
-      sonarQubeRunAnalysis: true
-      sqMavenPluginVersionChoice: 'latest'
-    displayName: Build/Analyze ${{ parameters.microservice }} --> Task
-# End of Maven-specific Analysis steps --> if
+- task: 'Maven@3'
+  inputs:
+    mavenPomFile: ${{ parameters.workDir }}pom.xml
+    publishJUnitresults: true
+    testResultsFiles: '**/surefire-reports/TEST-*.xml'
+    codeCoverageToolOption: 'JaCoCo'
+    codeCoverageClassFilesDirectories: ${{ parameters.coveragePath }}
+    javaHomeOption: 'JDKVersion'
+    jdkVersionOption: '1.11'
+    mavenVersionOption: 'Default'
+    mavenOptions: '-Xmx3072m'
+    mavenAuthenticateFeed: false
+    effectivePomSkip: false
+    sonarQubeRunAnalysis: true
+    sqMavenPluginVersionChoice: 'latest'
+  displayName: Build/Analyze ${{ parameters.microservice }} --> Task
 
 - task: SonarCloudPublish@1
   condition: always()
   inputs:
     pollingTimeoutSec: '300'
   displayName: Publish Analysis of ${{ parameters.microservice }} --> Task
 
-- script: |
-    echo $PARAMETERS_DISCORDCHANNEL
-    echo $PARAMETERS_DISCORDKEY
-  displayName: show discord info
-  
 - task: 'ado-discord-webhook@1' # <-- See external requirements note above
```

Comment:
duplication could be reduced here with templating, i think it's worth it but it's debatable

Reply:
I thought about having a template for notifications, but instead, I decided that we could have different notifications tailored to different purposes.  For example, a notification about analysis should link to the report, but a notification about a pending deployment approval would instead have a link to the page where it can be approved instead.  Therefore, I opted not to include the discord notification as a template, since it is only one task and I want to be able to change too many things, including the channelId, webhookKey, name, avatar, and message content... so basically everything, such that my template would end up being almost a complete redux of the task itself.  Other than going to the same channel and having a few common elements in the content section, even my PR version of the analysis is substantially different from the vanilla version.